pandas/_libs/tslibs/timestamps.pyx

```diff
@@ -379,6 +379,8 @@ class Timestamp(_Timestamp):
         _date_attributes = [year, month, day, hour, minute, second,
                             microsecond, nanosecond]
 
+        _non_ts_attributes = [freq, tz, unit, tzinfo] + _date_attributes
```

Comment:
I suggest we check for emtpy kwargs similar to the way we check for empty date attributes.